### PR TITLE
Respect authorisation request opt-out on workspace start

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
If user rejects an scm provider authorisation request while creating or starting existed workspace store the name of the scm provider in the `workspace-preferences` config-map. The workspace create/start step must proceed without token fetch step. If user creates another workspace or starts existed workspace from an scm provider which name is stored in the config-map, do not ask the authorisation as it was already rejected once.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22499

### How to test this PR?

1. Deploy che from the PR image: `quay.io/eclipse/che-server:pr-576`
2. Setup [GitHub oauth flow](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-github/)
3. Start a workspace from a public GitHub repository url.
4. When the authorisation page opens, click the `cancel` button.
5. When workspace starts, stop it and then start it, see: workspace starts without asking the authorisation.
6. Delete the workspace and create it again, see: workspace starts without asking the authorisation.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesAuthorisationRequestManager.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    HashSet<String> fromJson = getSkipAuthorisationValues();
+    fromJson.add(scmProviderName);
+
+    configMap.setData(Map.of(SKIP_AUTHORISATION_MAP_KEY, fromJson.toString()));
+
+    patchConfigMap(configMap);
+  }
+
+  @Override
+  public void remove(String scmProviderName) {
```
Comment (amisevsk) [inline infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesAuthorisationRequestManager.java:69]: If the user denies the OAuth request, is there any flow for "I changed my mind"? It seems like with this change Che will silently never ask again for OAuth access. This is probably fine for now, but maybe something we should think about in the future.

**Review by amisevsk — APPROVED** [review]
Code generally looks good to me, though I haven't been able to test the changes.

File: infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesAuthorisationRequestManager.java (hunk 1 above)
Comment (vinokurig) [reply infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesAuthorisationRequestManager.java:69]: This is going to be solved by the next iteration in the dashboard part. We are going to have an additional checkbox, or something like that, for each item in the `Git Services` tab of the user preferences page.

**Comment by artaleks9** [comment]
Verified according to steps in the `How to test this PR?` - the functionality works properly.

**Review by artaleks9 — APPROVED** [review]

File: infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesAuthorisationRequestManager.java (hunk 1 above)
Comment (vinokurig) [reply infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesAuthorisationRequestManager.java:69]: The corresponding dashboard issue has been created https://github.com/eclipse/che/issues/22600

**Review by ibuziuk — APPROVED** [review]

**Pull request merged**